src/webpack/patchWebpack.ts

```diff
@@ -1,353 +1,536 @@
+        const proxiedModuleFactories = new Proxy(originalModules, moduleFactoriesHandler);
+        /*
+        If Webpack ever decides to set module factories using the variable of the modules object directly, instead of wreq.m, switch the proxy to the prototype
+        Reflect.setPrototypeOf(originalModules, new Proxy(originalModules, moduleFactoriesHandler));
+        */
+
+        define(this, "m", { value: proxiedModuleFactories });
     }
 });
 
-function patchPush(webpackGlobal: any) {
-    function handlePush(chunk: any) {
+const moduleFactoriesHandler: ProxyHandler<AnyWebpackRequire["m"]> = {
+    /*
+    If Webpack ever decides to set module factories using the variable of the modules object directly instead of wreq.m, we need to switch the proxy to the prototype
+    and that requires defining additional traps for keeping the object working
+
+    // Proxies on the prototype dont intercept "get" when the property is in the object itself. But in case it isn't we need to return undefined,
+    // to avoid Reflect.get having no effect and causing a stack overflow
+    get: (target, p, receiver) => {
+        return undefined;
+    },
+    // Same thing as get
+    has: (target, p) => {
+        return false;
+    },
+    */
+
+    // The set trap for patching or defining getters for the module factories when new module factories are loaded
+    set: (target, p, newValue, receiver) => {
+        // If the property is not a number, we are not dealing with a module factory
+        if (Number.isNaN(Number(p))) {
+            return define(target, p, { value: newValue });
+        }
+
+        if (updateExistingFactory(target, p, newValue)) {
+            return true;
+        }
+
+        notifyFactoryListeners(newValue);
+        defineModulesFactoryGetter(p, Settings.eagerPatches && shouldPatchFactories ? wrapAndPatchFactory(p, newValue) : newValue);
+
+        return true;
+    }
+};
+
+/**
+ * Update a factory that exists in any Webpack instance with a new original factory.
+ *
+ * @target The module factories where this new original factory is being set
+ * @param id The id of the module
+ * @param newFactory The new original factory
+ * @param ignoreExistingInTarget Whether to ignore checking if the factory already exists in the moduleFactoriesTarget
+ * @returns Whether the original factory was updated, or false if it doesn't exist in any Webpack instance
+ */
+function updateExistingFactory(moduleFactoriesTarget: AnyWebpackRequire["m"], id: PropertyKey, newFactory: AnyModuleFactory, ignoreExistingInTarget: boolean = false) {
+    let existingFactory: TypedPropertyDescriptor<AnyModuleFactory> | undefined;
+    let moduleFactoriesWithFactory: AnyWebpackRequire["m"] | undefined;
+    for (const wreq of allWebpackInstances) {
+        if (ignoreExistingInTarget && wreq.m === moduleFactoriesTarget) continue;
+
+        if (Reflect.getOwnPropertyDescriptor(wreq.m, id) != null) {
```

Comment:
should be equivalent and better

```suggestion
        if (Object.hasOwn(wreq.m, id)) {
```